src/sqllogictest/src/runner.rs

```diff
@@ -1667,6 +1694,49 @@ fn print_sql<'a>(stdout: &'a dyn WriteFmt, sql: &str) {
     writeln!(stdout, "{}", crate::util::indent(sql, 4))
 }
 
+/// Regular expressions for matching error messages that should force a plan failure
+/// in an inconsistent view outcome into a warning if the corresponding query succeeds.
+const INCONSISTENT_VIEW_OUTCOME_WARNING_REGEXPS: [&str; 8] = [
+    // The following are unfixable errors in indexed views given our
+    // current constraints.
+    "cannot materialize call to",
+    "SHOW commands are not allowed in views",
+    "cannot create view with unstable dependencies",
+    "cannot use wildcard expansions or NATURAL JOINs in a view that depends on system objects",
+    "no schema has been selected to create in",
+    r#"system schema '\w+' cannot be modified"#,
+    // NOTE(vmarcos): Column ambiguity that could not be eliminated by our
+    // currently implemented syntactic rewrites is considered unfixable.
+    // In addition, if some column cannot be dealt with, e.g., in `ORDER BY`
+    // references, we treat this condition as unfixable as well.
+    r#"column "[\w\?]+" specified more than once"#,
+    r#"column "(\w+\.)?\w+" does not exist"#,
```

Comment:
Probably need to add `permission denied for SCHEMA`

Reply:
This is a good suggestion, which I just incorporated now. It is odd that locally I got different errors when running `schemas.slt`. Anyhow, adding the regular expression to catch also these permission denied errors allows us to pass `test/sqllogictest/privilege_grants.slt`. So one more file also included now.

Reply:
This is because we enable RBAC in the CI with MZ_SYSTEM_PARAMETER_DEFAULT since https://github.com/MaterializeInc/materialize/pull/20451

Reply:
Ah, this explains it, thanks! We have now a green slow SLT build: https://buildkite.com/materialize/sql-logic-tests/builds/5616. I'll get some final comments from Joe, but this should be pretty much ready to go. Enjoy your leave!

Reply:
> I'll get some final comments from Joe

Sorry, this has been on my TODO list, I'll take a look today.

Reply:
No worries, thanks!

Reply:
What cluster and schema do you create the views and indexes on? You can probably get around the permission errors by always creating them on a specific cluster and schema that you know you'll have permissions on. Alternatively, you can do the auto index select work as `mz_system` which is a superuser and will never get a permission error.

Reply:
Right now, the view creation and indexing statements are just issued directly through the same `tokio_postgres::Client` as the other commands in the SLT, without any changes of user/cluster/schema. The reasoning is that we'd like to, if possible, also match errors. In theory, upgrading to `mz_system`, for example, could make some queries succeed where the test expects them to error out. In this case, the outcome between view-based execution and query-based execution would be considered inconsistent, leading to an SLT failure.

Reply:
A refinement of your idea could be to only upgrade (say to `mz_system`) queries where the outcome is expected to be successful. I suggest that we monitor if maintaining the allow list for warnings gets too demanding, and if so, we can try this solution out.